Mathlib/Algebra/FreeMonoid/Basic.lean

```diff
@@ -407,4 +415,31 @@ theorem length_reverse {a : FreeMonoid α} : a.reverse.length = a.length :=
 
 end Reverse
 
+section IsomorphicTypes
+
+variable {α β : Type*} (e : α ≃ β)
+
+/-- free monoids over isomorphic types are isomorphic -/
+@[to_additive "if two types are isomorphic, the additive free monoids over those types are
+isomorphic"]
+def congr_iso : FreeMonoid α ≃* FreeMonoid β :=
+  MulEquiv.mk' ⟨FreeMonoid.map e.toFun, FreeMonoid.map e.invFun, fun _ => map_invFun_map_toFun_eq e,
+    fun _ => map_toFun_map_invFun_eq e⟩ (by simp [map_mul])
+
+/-- given an isomorphism between α and β, convert a relation predicate on FreeMonoid α to
+have an underlying type of β -/
+@[to_additive "given an isomorphism between α and β, convert a relation predicate to
+have an underlying type of β"]
+def map_rel (rel : FreeMonoid α → FreeMonoid α → Prop) : FreeMonoid β → FreeMonoid β  → Prop :=
+  fun a b ↦ rel (.congr_iso e.symm a) (.congr_iso e.symm b)
+
+/-- given an isomorphism between α and β, pull back a relation predicate with underlying type β to
+one with underlying type α -/
+@[to_additive "given an isomorphism between α and β, pull back a relation predicate with underlying
+type β to one with underlying type α "]
+def comap_rel (rel : FreeMonoid β → FreeMonoid β → Prop) : FreeMonoid α → FreeMonoid α → Prop :=
```

Comment:
```suggestion
def comapRel (rel : FreeMonoid β → FreeMonoid β → Prop) : FreeMonoid α → FreeMonoid α → Prop :=
```
Could you please generalize this to a function `f : \a \to \b` using `FreeMonoid.map`? Because you don't need an equiv between α and β for this definition; a function is enough.

Reply:
oy, you're totally right. fixed!